### ISSUE TITLE
Add support for CloudFormation Parameters

### ISSUE THREAD

**Issue by smithhannahm** [awslabs/goformation#220]
There isn't any out of the box support for creating template parameters. This is somewhat easily worked around by creating a local parameter struct, but it would be nice to have that included in the package.

``` go
type parameter struct {
	Type                  string   `json:"Type"`
	Description           string   `json:"Description,omitempty"`
	Default               string   `json:"Default,omitempty"`
	AllowedPattern        string   `json:"AllowedPattern,omitempty"`
	AllowedValues         []string `json:"AllowedValues,omitempty"`
	ConstraintDescription string   `json:"ConstraintDescription,omitempty"`
	MaxLength             int      `json:"MaxLength,omitempty"`
	MinLength             int      `json:"MinLength,omitempty"`
	MaxValue              float64  `json:"MaxValue,omitempty"`
	MinValue              float64  `json:"MinValue,omitempty"`
	NoEcho                bool     `json:"NoEcho,omitempty"`
}
```

**Comment by PaulMaddox** [comment]
Hi - do you want to create a PR for this? I agree it would be good to have.

**Comment by th3noname** [comment]
I went ahead and created a PW for this, since i would like to have this too.

**Comment by xrn** [comment]
@PaulMaddox this issue can be probably closed :)

**Comment by xrn** [comment]
@PaulMaddox I am reviewing issues again and IMO it can be closed as solved in PR #259 :)

**Comment by PaulMaddox** [comment]
Thanks all :)
